refactor/src/Cage.py

```diff
@@ -136,7 +206,10 @@ def update(self, cur_date: dt.datetime, step_size: int, pressure: int) -> tuple:
 
         # Mating events that create eggs
         delta_avail_dams, delta_eggs = self.do_mating_events()
-        egg_distrib, hatch_date = self.get_egg_batch(cur_date, delta_eggs)
+        avail_dams_batch = DamAvailabilityBatch(cur_date + dt.timedelta(days=self.cfg.dam_unavailability),
+                                                delta_avail_dams)
+
+        new_egg_batch = self.get_egg_batch(cur_date, delta_eggs)
```

Comment:
This a bigger issue not limited to this line: there is a reason I have removed EggBatch in this line in the last PR - the egg queue should not be updated in the main Cage update.

The new offspring is collected on the driver level and added to individual farms and cages after the main farm updates are completed. This is because the egg movement requires access to different farms - and the main farm/cage update doesn't, so it can be multiprocessed in the future. 

The interfarm flow is as follows: the cages update and return the offspring, the farms update and return all the offspring from cages then the driver collects that information and disperses the offspring between the farms (ie updates the arrivals queue). Note that this includes same farm movement, that is "normal" offspring that was produced and will live in the same farm.

I am open to changing this but please note that at the moment this introduces a bug where offspring is added twice (once in main update and once in disperse update).